From e2a6adeb0f6cf8ac589e0f93d14739658418e788 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Mon, 30 Apr 2018 13:11:37 +0300
Subject: [PATCH 142/706] enetc: ethtool: Add interface uncorrectable error
 frame drop count reg

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 5f8a2daae22c50a3b0c7bb4ccf1760d586fd8d15)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_ethtool.c | 2 +-
 drivers/net/ethernet/freescale/enetc/enetc_hw.h      | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_ethtool.c b/drivers/net/ethernet/freescale/enetc/enetc_ethtool.c
index e7828924c343..541a7dce8831 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_ethtool.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_ethtool.c
@@ -38,7 +38,7 @@ static const u32 enetc_si_regs[] = {
 	ENETC_SIMR, ENETC_SIPMAR0, ENETC_SIPMAR1, ENETC_SICBDRMR,
 	ENETC_SICBDRSR,	ENETC_SICBDRBAR0, ENETC_SICBDRBAR1, ENETC_SICBDRCIR,
 	ENETC_SICBDRCISR, ENETC_SICBDRLENR, ENETC_SICAPR0, ENETC_SICAPR1,
-	ENETC_PRFSCAPR
+	ENETC_SIUEFDCR, ENETC_PRFSCAPR
 };
 
 static const u32 enetc_txbdr_regs[] = {
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index ac6a55747d67..8c51b4b8973a 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -117,6 +117,8 @@ static inline u32 enetc_vsi_set_msize(u32 size)
 #define ENETC_SIMSITRV(n) (0xB00 + (n) * 0x4)
 #define ENETC_SIMSIRRV(n) (0xB80 + (n) * 0x4)
 
+#define ENETC_SIUEFDCR	0xe28
+
 #define ENETC_SIRFSCAPR	0x1200
 
 /** SI BDR sub-blocks, n = 0..7 */
-- 
2.17.1

